python/doc/examples/calibration/bayesian_calibration/plot_rwmh_python_distribution.py

```diff
@@ -0,0 +1,174 @@
+        self.beta = beta
+        self.eta = eta
+    
+    def getRange(self):
+        return ot.Interval([0, 0], [1, 1], [True]*2, [False, True])    
+
+    def computeLogPDF(self, x):
+        log_pdf = -( x[0] / self.eta )**self.beta
+        log_pdf += ( self.beta - 1 ) * np.log( x[0] / self.eta ) * x[1]
+        log_pdf += np.log( self.beta / self.eta ) * x[1]
+        return log_pdf
+    
+    def setParameter( self, parameter ):
+        self.beta = parameter[0]
+        self.eta = parameter[1]
+
+    def getParameter( self ):
+        return [self.beta, self.eta]
+
+# %%
+# Convert to ot.Distribution 
+
+# %%
+
+conditional = ot.Distribution( CensoredWeibull() )
+
+
+# %%
+# Observations
+
+
+# %%
+
+Tobs = np.array([4380, 1791, 1611, 1291, 6132, 5694, 5296, 4818, 4818, 4380])
+fail = np.array( [True]*4+[False]*6 )
+x = ot.Sample( np.vstack((Tobs, fail)).T )
+
+
+# %%
+# Define a uniform prior distribution for :math:`\beta` and a Gamma distribution for :math:`\eta`
+# 
+
+# %%
+
+beta_min, beta_max = 0.5, 3.8 
+a_eta, b_eta = 2, 2e-4 
+
+priorCopula = ot.IndependentCopula(2)# prior independence
+priorMarginals = []# prior marginals
+priorMarginals.append(ot.Uniform(beta_min, beta_max))# uniform prior for beta
+priorMarginals.append(ot.Gamma(a_eta, b_eta))# Gamma prior pour eta
+prior=ot.ComposedDistribution( priorMarginals, priorCopula )
+prior.setDescription(['beta','eta'])
+
+
+
+# %%
+# Initial point of the Metropolis-Hastings algorithm : prior means
+# 
+
+# %%
+
+initialState = ot.Point([ 0.5*(beta_max - beta_min), a_eta / b_eta ])
+
+# %%
+# Random walk proposals: Normal steps, with standard deviation equal to roughly :math:`10\%` of the prior range or standard deviation
+# 
+
+# %%
+
+proposal=[]
+proposal.append( ot.Normal(0., 0.1 * ( beta_max - beta_min ) ) )
+proposal.append( ot.Normal(0., 0.1 * np.sqrt( a_eta / b_eta**2 ) ) )
+
+# %%
+# All set! Now launch the posterior sampling algorithm
+# 
+
+# %%
+
+RWMHsampler = ot.RandomWalkMetropolisHastings(prior, conditional, x, initialState, proposal)
+strategy = ot.CalibrationStrategyCollection(2)
+RWMHsampler.setCalibrationStrategyPerComponent(strategy)
+RWMHsampler.setVerbose(True)
+sampleSize = 10000
+sample = RWMHsampler.getSample(sampleSize)
+# compute acceptance rate
+print("Acceptance rate: %s"%(RWMHsampler.getAcceptanceRate()))
+
+# %%
+# Plot prior to posterior marginal plots
+# 
+
+# %%
+
+kernel = ot.KernelSmoothing()
+posterior = kernel.build(sample)
+fig = plt.figure(figsize=(12, 4))
+for parameter_index in range(2):
+    graph = posterior.getMarginal(parameter_index).drawPDF()
+    priorGraph = prior.getMarginal(parameter_index).drawPDF()
+    priorGraph.setColors(['blue'])
+    graph.add(priorGraph)
+    graph.setLegends(['Posterior', 'Prior', 'MLE'])
+    ax = fig.add_subplot(1, 2, parameter_index+1)
+    _ = View(graph, figure=fig, axes=[ax])
+    plt.legend(['Posterior', 'Prior'])
+    
+_ = fig.suptitle("Bayesian inference")
+    
```

Comment:
```suggestion
kernel = ot.KernelSmoothing()
posterior = kernel.build(sample)
grid = ot.GridLayout(1, 2)
grid.setTitle('Bayesian inference')
for parameter_index in range(2):
    graph = posterior.getMarginal(parameter_index).drawPDF()
    priorGraph = prior.getMarginal(parameter_index).drawPDF()
    graph.add(priorGraph)
    graph.setColors(ot.Drawable_BuildDefaultPalette(2))
    graph.setLegends(['Posterior', 'Prior'])
    grid.setGraph(0, parameter_index, graph)
_ = View(grid)
```
La classe GridLayout répond au besoin sans avoir à passer par matplotlib. Je me permets aussi de modifier les couleurs pour utiliser la palette par défaut, qui me paraît plus esthétique que le couple bleu/rouge. Voici ce que ça donne : 
![image](https://user-images.githubusercontent.com/31989332/112998434-539a8e00-916e-11eb-99a9-f84f7a7124c0.png)